aiida/engine/processes/calcjobs/tasks.py

```diff
@@ -406,7 +407,7 @@ async def execute(self) -> plumpy.process_states.State:  # type: ignore[override
             else:
                 logger.warning(f'killed CalcJob<{node.pk}> but async future was None')
             raise
-        except (plumpy.process_states.Interruption, plumpy.futures.CancelledError):
+        except (plumpy.process_states.Interruption, plumpy.futures.CancelledError, asyncio.CancelledError):
```

Comment:
should `plumpy.futures.CancelledError` *be* `asyncio.CancelledError`?

I notice that `concurrent.futures` is used in kiwipy - where are we currently using `concurrent.futures` and where `asyncio` and what's the rationale behind this?
Also mentioning @muhrin

Reply:
> should `plumpy.futures.CancelledError` be `asyncio.CancelledError`

yes I think this should be the case

> where are we currently using concurrent.futures and where asyncio and what's the rationale behind this?

@muhrin can correct me if I am wrong, but we use `concurrent.futures.Future` when we are dealing with threads, and `asyncio.Future` for standard asynchronicity.
Before python 3.8, calling the `cancel()` method of either of these generated `plumpy.futures.CancelledError`, but now they generate different exception so, as I've mentioned (https://github.com/aiidateam/aiida-core/issues/4648#issuecomment-790975477), we should check again at how these are dealt with in kiwipy

Reply:
yep no problem